homeassistant/components/nina/config_flow.py

```diff
@@ -0,0 +1,171 @@
+        """Handle the initial step."""
+        errors: dict[str, Any] = {}
+
+        if self._async_current_entries():
+            return self.async_abort(reason="single_instance_allowed")
+
+        has_error: bool = False
+
+        if len(self._all_region_codes_sorted) == 0:
+            try:
+                nina: Nina = Nina()
+
+                self._all_region_codes_sorted = self.swap_key_value(
+                    await nina.getAllRegionalCodes()
+                )
+
+                self.split_regions()
+
+            except ApiError as err:
+                _LOGGER.warning("NINA setup error: %s", err)
+                errors["base"] = "cannot_connect"
+                has_error = True
+            except Exception as err:  # pylint: disable=broad-except
+                _LOGGER.exception("Unexpected exception: %s", err)
+                errors["base"] = "unknown"
+                return self.async_abort(reason="unknown")
+
+        if user_input is not None and not has_error:
+            config: dict[str, Any] = user_input
+
+            config[CONF_REGIONS] = []
+
+            for group in CONST_REGIONS:
+                if group_input := user_input.get(group):
+                    config[CONF_REGIONS] += group_input
+
+            if len(config[CONF_REGIONS]) > 0:
+                tmp: dict[str, Any] = {}
+
+                for reg in config[CONF_REGIONS]:
+                    tmp[self._all_region_codes_sorted[reg]] = reg.split("_", 1)[0]
+
+                compact: dict[str, Any] = {}
+
+                for key, val in tmp.items():
+                    if val in compact:
+                        compact[val] = f"{compact[val]} + {key}"
+                        break
+                    compact[val] = key
+
+                config[CONF_REGIONS] = compact
+
+                return self.async_create_entry(title="NINA", data=config)
+
+            errors["base"] = "no_selection"
+
+        return self.async_show_form(
+            step_id="user",
+            data_schema=vol.Schema(
+                {
+                    vol.Optional(CONST_REGION_A_TO_D): cv.multi_select(
+                        self.regions[CONST_REGION_A_TO_D]
+                    ),
+                    vol.Optional(CONST_REGION_E_TO_H): cv.multi_select(
+                        self.regions[CONST_REGION_E_TO_H]
+                    ),
+                    vol.Optional(CONST_REGION_I_TO_L): cv.multi_select(
+                        self.regions[CONST_REGION_I_TO_L]
+                    ),
+                    vol.Optional(CONST_REGION_M_TO_Q): cv.multi_select(
+                        self.regions[CONST_REGION_M_TO_Q]
+                    ),
+                    vol.Optional(CONST_REGION_R_TO_U): cv.multi_select(
+                        self.regions[CONST_REGION_R_TO_U]
+                    ),
+                    vol.Optional(CONST_REGION_V_TO_UE): cv.multi_select(
+                        self.regions[CONST_REGION_V_TO_UE]
+                    ),
```

Comment:
Consider using dict comprehension

```suggestion
                    **{vol.Optional(region): cv.multi_select(
                        self.regions[region]
                    ) for region in CONST_REGIONS}
```